### PR TITLE
Fix arrows requiring a radius to be visible

### LINKED ISSUE #1708: Logging Arrows without radius doesn't show anything
Minimal repro:
```
fn run(mut session: rerun::Session) -> Result<(), rerun::MsgSenderError> {
    rerun::MsgSender::new("my_points")
        .with_component(&[rerun::components::Arrow3D {
            origin: rerun::components::Vec3D::new(0.0, 0.0, 0.0),
            vector: rerun::components::Vec3D::new(0.0, 10.0, 1.0),
        }])?
        .send(&mut session)
}

fn main() {
    let session = rerun::Session::init("arrow", true);
    session.spawn(run).unwrap();
}
```

Need to add `.with_component(&[rerun::components::Radius(1.0)])?` to see anything. Also happens in python!

Had a brief look in our code: I think the problem is that `LineStripFlags::get_triangle_cap_tip_length` doesn't deal with auto size correctly, but haven't given it much thought. In any case the issue is bound to be in `Arrows3DPart`

### PR DESCRIPTION
**Pull request by Wumpf**
Fixes #1708

.. that is the immediately user-facing problem this solves. But in trying to solve this properly I had to make the line renderer a bit more predictable: Previously, line caps would extend the line. Now they stay within the bounds of the previously determined line positions!

2D Demo before - note how caps make the lines longer:
image

and after - all the same length now:
image

rounded rectangles are still rounded (requires outward extending the rounded cap):
image

What an outline around it looks like (clock example):
image

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)

### CODE REVIEW

**Comment by Wumpf** [comment]
oops. not actually the same length yet on all combination. Thanks for pointing that out @nikolausWest 😅

**Comment by Wumpf** [comment]
I actually broke rectangles I noticed a bit too late. I have now an option to configure the outward extensibility which addresses both that issue and your comment :)
(at least partially, as I don't allow mixing that yet, could easily extend it though)
View doesn't use rounded caps anywhere else yet

File: crates/re_renderer/shader/lines.wgsl [hunk 1]
```diff
@@ -41,7 +41,8 @@ const CAP_END_TRIANGLE: u32 = 1u;
 const CAP_END_ROUND: u32 = 2u;
 const CAP_START_TRIANGLE: u32 = 4u;
 const CAP_START_ROUND: u32 = 8u;
-const NO_COLOR_GRADIENT: u32 = 16u;
+const CAP_EXTEND_OUTWARDS: u32 = 16u;
```
Comment (emilk) [inline crates/re_renderer/shader/lines.wgsl:44]: Can we have one per side instead for full control?

File: crates/re_viewer/src/ui/view_spatial/scene/scene_part/arrows3d.rs [hunk 2]
```diff
@@ -67,9 +67,7 @@ impl Arrows3DPart {
             let origin = glam::Vec3::from(origin);
 
             let radius = radius.map_or(Size::AUTO, |r| Size(r.0));
-            let tip_length = LineStripFlags::get_triangle_cap_tip_length(radius.0);
-            let vector_len = vector.length();
-            let end = origin + vector * ((vector_len - tip_length) / vector_len);
+            let end = origin + vector;
```
Comment (emilk) [inline crates/re_viewer/src/ui/view_spatial/scene/scene_part/arrows3d.rs:70]: it would be nice to give arrows a rounded base while we're at it, but then we need something like `CAP_START_EXTEND_OUTWARDS`

Comment (Wumpf) [reply crates/re_viewer/src/ui/view_spatial/scene/scene_part/arrows3d.rs:70]: let's do that then, super easy now :)

**Comment by Wumpf** [comment]
image
there we go!
overlapping could still be handled better, but let's not go there now

File: crates/re_renderer/shader/lines.wgsl [hunk 3]
```diff
@@ -200,6 +214,29 @@ fn vs_main(@builtin(vertex_index) vertex_idx: u32) -> VertexOut {
...
+    }
+    if !has_any_flag(strip_data.flags, CAP_START_EXTEND_OUTWARDS) {
+        var neighbor_cap_flags = currently_active_flags;
+        if !is_at_quad_end && pos_data_current.strip_index != pos_data_quad_before.strip_index {
+            neighbor_cap_flags |= strip_data.flags & (CAP_START_TRIANGLE | CAP_START_ROUND);
+        }
+        if has_any_flag(neighbor_cap_flags, CAP_START_TRIANGLE | CAP_START_ROUND) {
+            center_position += quad_dir * cap_length(neighbor_cap_flags, strip_radius);
+        }
+    }
```
Comment (emilk) [inline crates/re_renderer/shader/lines.wgsl:236]: Can't we simplify all this quite a bit?

```suggestion
    if !has_any_flag(strip_data.flags, CAP_START_EXTEND_OUTWARDS) &&
       !is_at_quad_end && pos_data_current.strip_index != pos_data_quad_before.strip_index {
        var cap_length =
            ((strip_data.flags & CAP_START_ROUND) != 0) as f32 +
            ((strip_data.flags & CAP_START_TRIANGLE) != 0) as f32 * 4.0;
        center_position += quad_dir * cap_length * strip_radius;
    }
```

**Review by emilk — APPROVED** [review]
Looks good, but I think the code could be simplified quite a bit!

**Comment by Wumpf** [comment]
thanks for the input! Most of the stuff suggested doesn't work as-is because of some detail (typically on how cap triangles are flagged) but that does stress the point that this got a bit too hard to follow again. I'll try to make it simpler!
